stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -888,6 +910,49 @@ internal class ChannelLogic(
             }
     }
 
+    public fun toChannel(): Channel = mutableState.toChannel()
+
+    internal fun replyMessage(repliedMessage: Message?) {
+        mutableState._repliedMessage.value = repliedMessage
+    }
+
+    internal suspend fun sendMessage(message: Message) = messageSendingService.sendNewMessage(message)
+
+    /**
+     * Cancels ephemeral Message.
+     * Removes message from the offline storage and memory and notifies about update.
+     */
+    internal suspend fun cancelEphemeralMessage(message: Message): Result<Boolean> {
```

Comment:
`sendMessage`, `handleSendMessageSuccess`, `handleSendMessageFail` belongs to `SendMessageListenerImpl`. `observeAttachmentsForMessage`, `cancelEphemeralMessage`  belong to `MessageSendingService`. `selectMessage` is not being used and doesn't seam to be necessary as `RepositoryFacade` already does that.